flytekit/core/type_engine.py

```diff
@@ -2176,9 +2192,7 @@ def generate_attribute_list_from_dataclass_json(schema: dict, schema_name: typin
     return attribute_list
 
 
-def convert_marshmallow_json_schema_to_python_class(
-    schema: dict, schema_name: typing.Any
-) -> Type[dataclasses.dataclass()]:  # type: ignore
+def convert_marshmallow_json_schema_to_python_class(schema: dict, schema_name: typing.Any) -> type:
```

Comment:
https://github.com/python/cpython/issues/102699 is a discussion about having a baseclass for dataclasses that could be used here instead of `Type[dataclasses.dataclass()]` which is invalid.